app/repository/account_repository.go

```diff
@@ -0,0 +1,8 @@
+package repository
+
+import "github.com/jirlon/digitalbank/app/entities"
+
+// Defines the interface for account persistence operations.
+type AccountRepository interface {
+	SaveAccount(account *entities.Account) error
```

Comment:
Não utilizar ponteiro aqui também
```suggestion
	SaveAccount(account entities.Account) error
```